app/views/registrations/_item_row.html.erb

```diff
@@ -34,6 +34,10 @@
 <div class="row mb-3">
   <%= item_form.label :barcode, class: 'col-sm-2 col-form-label' %>
   <div class="col-sm-10">
-    <%= item_form.text_field :barcode, class: 'form-control' %>
+    <%= item_form.text_field :barcode, class: 'form-control',
+    pattern: '^(36105[0-9]{9}|2050[0-9]{7}|245[0-9]{8}|[0-9]+-[0-9]+)',
+    data: {
+      action: "change->registration#validateBarcode invalid->registration#displayValidation",
```

Comment:
There is no validateBarcode method on registration controller.

Reply:
right you are.

Reply:
there is now.  Though I'm not sure if there should be.